libs/http/src/middleware/token_auth.cpp

```diff
@@ -0,0 +1,52 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/logging.hpp"
+#include "http/middleware/token_auth.hpp"
+
+#include <memory>
+
+namespace fetch {
+namespace http {
+namespace middleware {
+
+HTTPServer::RequestMiddleware TokenAuth(byte_array::ConstByteArray const &token, uint32_t level,
```

Comment:
I think that is makes the most sense to pass in a reference to a class interface here like:

```
class AuthenticationInterface
{
public:
  virtual uint32_t AuthenticateToken(ConstByteArray const &token) = 0;
};
```
We can provide very simple concrete versions of this initially, but allows this to be easily extended in the future.